frontend/packages/ceph-storage-plugin/src/components/create-storage-system/reducer.ts

```diff
@@ -125,12 +121,80 @@ export type LocalVolumeSet = {
   chartNodes: Set<string>;
 };
 
+const setDeployment = (
+  prevState: WizardState,
+  newState: WizardState,
+  deploymentType: DeploymentType,
+) => {
+  /*
+   * Wizard state should be reset when a new deployment type is selected
+   * in order to avoid stale state collisions since each deployment mode
+   * has its own supported configuration.
+   *
+   * Its not required if the user has not visited more than first step.
+   */
+  if (prevState.stepIdReached !== 1) {
+    const { type } = prevState.backingStorage;
+    return {
+      ...initialState,
+      storageClass:
+        type === BackingStorageType.EXISTING ? prevState.storageClass : initialState.storageClass,
+      backingStorage: {
+        ...prevState.backingStorage,
+        deployment: deploymentType,
+      },
+    };
+  }
+
+  newState.backingStorage.deployment = deploymentType;
+  return newState;
+};
+
+const setBackingStorageType = (
+  prevState: WizardState,
+  newState: WizardState,
```

Comment:
```suggestion
```
`newState` is a clone of `prevState` why are we passing two exact values? Can you not use `Object.assign` to update values.

Reply:
same as below